### ISSUE TITLE
Add Feature for Easy Access to Keywords in Docs Glossary

### ISSUE THREAD

**Issue by mboukhalfa** [google/docsy#1997]
As I work on localizing Kubernetes documentation into Arabic using the Docsy theme, I've encountered a significant opportunity to improve the readability of localized documents.

When translating documentation from English to another language, one common challenge is finding suitable translations for keywords. Often, the translated term is either unfamiliar or lacks a unified translation, leading to the retention of the English term or the awkward usage of both the translated term and its English counterpart in parentheses.

**To address this issue, I propose the following enhancement:**

I suggest implementing a feature that allows users to easily access the English term when they encounter an unfamiliar translated term. One approach could involve enabling users to hover over the translated term, triggering a clean and unobtrusive tooltip display of the corresponding English term, similar to [Bootstrap tooltips](https://getbootstrap.com/docs/5.0/components/tooltips/).

**How Docsy can implement this:**
 
To streamline this process with Docsy, we could leverage a glossary for localization. By creating a glossary that links English terms with their translations, we could utilize shortcodes in Docsy to automatically generate tooltips for terms listed in the glossary. This approach would not only facilitate comprehension for users encountering unfamiliar terms but also provide an efficient means for users to access glossary definitions without leaving the current page.

I envision a straightforward process where I provide a glossary in JSON format and invoke a simple Docsy function. This function would generate tooltips for the keywords in the glossary that exist in a selected text.

Implementing this feature would greatly enhance development with Docsy and the user experience of localized documentation, ensuring clarity.

**Comment by fekete-robert** [comment]
For example here, in the second paragraph: https://falco.org/docs/getting-started/

**Comment by mboukhalfa** [comment]
> For example here, in the second paragraph: https://falco.org/docs/getting-started/

Yeah exactly what I had in mind I just want to implement this to link translated terms with the English ones so when if the reader come across unfamiliar translated term can easily refer to the English one

**Comment by sftim** [comment]
We already have [glossary tooltips](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#glossary), but only where the author has added them. We could add a similar mechanism for providing hints to people reading localized pages.

I wouldn't use JSON for that though.

**Comment by chalin** [comment]
- Pre-0.17.0 triage campaign: #2717
- No follow-through since 2024; the falco-docs approach referenced in-thread remains available site-side.
